test/integration/Cython_0001-Pyston-change-we-don-t-support-custom-traceback-entr.patch

```diff
@@ -1,14 +1,42 @@
-From 0d1aaa649bd47a3f78e573eb4e5b9e35f9aa02d3 Mon Sep 17 00:00:00 2001
+From 27b69179b90851da187d698a0817b0b8190a0449 Mon Sep 17 00:00:00 2001
 From: Marius Wachtler <undingen@gmail.com>
 Date: Tue, 9 Jun 2015 19:26:44 +0200
 Subject: [PATCH] Pyston change: we don't support custom traceback entries yet
 
 ---
- Cython/Compiler/ModuleNode.py |  8 ++++++--
- Cython/Utility/Exceptions.c   |  7 ++++++-
- Cython/Utility/Generator.c    | 31 +++++++++++++++++++++----------
- 3 files changed, 33 insertions(+), 13 deletions(-)
+ Cython/Compiler/ExprNodes.py    |  9 +++++++++
+ Cython/Compiler/ModuleNode.py   |  8 ++++++--
+ Cython/Utility/CythonFunction.c |  4 +++-
+ Cython/Utility/Exceptions.c     |  7 ++++++-
+ Cython/Utility/Generator.c      | 41 +++++++++++++++++++++++++++++------------
+ 5 files changed, 53 insertions(+), 16 deletions(-)
 
+diff --git a/Cython/Compiler/ExprNodes.py b/Cython/Compiler/ExprNodes.py
+index f99ec6e..7ab41f3 100644
+--- a/Cython/Compiler/ExprNodes.py
++++ b/Cython/Compiler/ExprNodes.py
+@@ -7784,12 +7784,21 @@ class PyCFunctionNode(ExprNode, ModuleNameMixin):
+ 
+     @classmethod
+     def from_defnode(cls, node, binding):
++        """
++        # Pyston change: dont't generate custom code objects because we don't support them currently
+         return cls(node.pos,
+                    def_node=node,
+                    pymethdef_cname=node.entry.pymethdef_cname,
+                    binding=binding or node.specialized_cpdefs,
+                    specialized_cpdefs=node.specialized_cpdefs,
+                    code_object=CodeObjectNode(node))
++        """
++        return cls(node.pos,
++                   def_node=node,
++                   pymethdef_cname=node.entry.pymethdef_cname,
++                   binding=binding or node.specialized_cpdefs,
++                   specialized_cpdefs=node.specialized_cpdefs,
++                   code_object=None)
```

Comment:
This let's cython not generate ```PyCode_New``` calls. My previous workaround was to replace all callsites inside cython with ```None``` but as you pointed out that's ugly. That's now my slightly cleaner workaround. The problem is that supporting the ```PyCode_New``` function is hard because it tries to create a code object for all python functions which cython translates to c. I think the reason is that it need's it in order to generate backtraces. But we have them currently already disabled inside cython so it should not make a difference for us currently.